### PR TITLE
/printing/str.py: Fixes https://github.com/sympy/sympy/issues/10035 ,fixed str printer to return re-creatable representations for sets   

### PR DESCRIPTION
**Pull request by ghost**
I have tried to fix the str printer so that it returns re-creatable representations of sets S.Reals, S.Complexes, S.Naturals. I have also updated the respective test files as well.

### CODE REVIEW

**Comment by achetan40** [comment]
One test is failing but I do not think that it has anything to do with my commit because it is failing in the master as well. 

The file is sympy/plotting/tests/test_plot.py

File: sympy/printing/pretty/tests/test_pretty.py [hunk 1]
```diff
@@ -5553,9 +5553,10 @@ def test_issue_7180():
 
 
 def test_pretty_Complement():
-    assert pretty(S.Reals - S.Naturals) == '(-oo, oo) \ Naturals()'
+
+    assert pretty(S.Reals - S.Naturals) == 'S.Reals \ S.Naturals'
     assert upretty(S.Reals - S.Naturals) == u'ℝ \ ℕ'
-    assert pretty(S.Reals - S.Naturals0) == '(-oo, oo) \ Naturals0()'
+    assert pretty(S.Reals - S.Naturals0) == 'S.Reals \ S.Naturals'
```
Comment (asmeurer) [inline sympy/printing/pretty/tests/test_pretty.py]: The pretty printing for reals should stay the same. Also, S.Naturals0 should not print as S.Naturals!

File: sympy/printing/str.py [hunk 2]
```diff
@@ -44,6 +44,16 @@ def emptyPrinter(self, expr):
         else:
             return str(expr)
 
+    def _print_Reals(self,expr):
+        return "S.Reals"
+
+    def _print_Complexes(self,expr):
+        return "S.Complexes"
+
+    def _print_Naturals(self,expr):
```
Comment (asmeurer) [inline sympy/printing/str.py]: The methods in this file are (mostly) in alphabetical order. You should add these in accordingly.

Comment (achetan40) [inline sympy/printing/str.py]: Thanks for the review. I will keep the alphabetical order in mind. Also for pretty printing the intervals, would shifting the function that prints intervals to pprint.py work?

**Comment by aktech** [comment]
Please have a look at https://github.com/sympy/sympy/wiki/Development-workflow#writing-pull-request-title-and-description

**Comment by gxyd** [comment]
@ghost can you please address the comments? Merge the master branch, and improve the PR title.

**Comment by gxyd** [comment]
Changes in this PR has already been merged in SymPy master. Done by PR #12112. Closing this.

**Pull request closed without merging**